org.lflang/src/org/lflang/generator/ts/TSActionGenerator.kt

```diff
@@ -0,0 +1,78 @@
+package org.lflang.generator.ts
+
+import org.lflang.generator.PrependOperator
+import org.lflang.lf.Action
+import org.lflang.lf.Reactor
+import org.lflang.lf.Type
+import org.lflang.lf.Value
+import java.util.*
+
+class TSActionGenerator (
+    private val tsGenerator: TSGenerator,
```

Comment:
It shouldn't be necessary to pass in a reference to the generator. However, looking at `CppActionGenerator`, I'm not sure where it gets `$targetType` from. @cmnrd could you explain this?

Reply:
I took a note as a TODO in the code. I'll try to figure out how to remove this dependency.